wallet/src/account/output_cache/mod.rs

```diff
@@ -151,4 +187,84 @@ impl OutputCache {
 
         utxos
     }
+
+    pub fn get_abandonable_transactions(&self) -> Vec<&WithId<Transaction>> {
+        self.txs
+            .values()
+            .filter_map(|tx| match tx {
+                WalletTx::Block(_) => None,
+                WalletTx::Tx(tx) => match tx.state() {
+                    TxState::Inactive => Some(tx.get_transaction_with_id()),
+                    TxState::Confirmed(_, _)
+                    | TxState::Conflicted(_)
+                    | TxState::InMempool
+                    | TxState::Abandoned => None,
+                },
+            })
+            .collect()
+    }
+
+    pub fn abandon_transaction(&mut self, tx_id: Id<Transaction>) -> WalletResult<()> {
+        let mut to_abandon = BTreeSet::new();
+        to_abandon.insert(OutPointSourceId::from(tx_id));
+
+        while let Some(outpoint_source_id) = to_abandon.pop_first() {
+            if let Some(descendants) = self.unconfirmed_descendants.remove(&outpoint_source_id) {
+                to_abandon.extend(descendants.into_iter())
+            }
+
+            match self.txs.entry(outpoint_source_id) {
+                Entry::Occupied(mut entry) => match entry.get_mut() {
+                    WalletTx::Block(_) => Err(WalletError::CannotFindTransactionWithId(tx_id)),
+                    WalletTx::Tx(tx) => match tx.state() {
+                        TxState::Inactive => {
+                            tx.set_state(TxState::Abandoned);
+                            for input in tx.get_transaction().inputs() {
+                                match input {
+                                    TxInput::Utxo(outpoint) => {
+                                        self.consumed.insert(outpoint.clone(), *tx.state());
+                                    }
+                                    TxInput::Account(_) => {
+                                        unimplemented!()
+                                    }
+                                }
+                            }
+                            Ok(())
+                        }
+                        state => Err(WalletError::CannotAbandonTransaction(*state)),
+                    },
+                },
+                Entry::Vacant(_) => Err(WalletError::CannotFindTransactionWithId(tx_id)),
+            }?;
+        }
+
+        Ok(())
+    }
+}
+
+// TODO: similar code from tx verifier
```

Comment:
You mean use the code from tx verifier? If yes, what's stopping you?

Reply:
It is not exactly the same and I was not sure if I should introduce the dependency.

Reply:
What's the difference? Can't one be a superset of the other?